yamux/src/connection.rs

```diff
@@ -609,7 +620,8 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Active<T> {
             frame
         };
         if let Some(f) = frame {
-            log::trace!("{}/{}: sending: {}", self.id, stream_id, f.header());
+            let pf = f.parse().expect("valid frame");
```

Comment:
It is quite unfortunate that we have all these panic paths now. I guess "parsing" using zerocopy is very cheap? Can we validate that a frame is valid when we initially construct it and re-parse it every time we need it in order to avoid this?